### PR TITLE
Create .gitignore

### PR DESCRIPTION
**Pull request by Pushplaybang**
First attempt at a standard starting .gitignore.  Will always need to be updated for specific projects.  These are some of the standard system bits that we should ignore.

Interestingly, bower suggest not ignoring bower components and there seem to be some debate. heres a good reference.
https://addyosmani.com/blog/checking-in-front-end-dependencies/

### CODE REVIEW

**Comment by ShaunEgan** [comment]
This guy could (_should?_) be a global ignore based on the developer's choice of tools/OS. Otherwise this could just grow and grow. Im not particularly phased though. :+1:

**Comment by SkippyZA** [comment]
Personally, I not a fan of these massive .gitignore files. The .gitignore in the project should only exclude project related artifacts that should not be committed to source control.

Any IDE and OS specific excludes should be done as a global .gitignore file on their machine.

Obviously there is the argument of 'at least this prevents them from being committed by accident', sure, but with code review, it shouldn't make it through. When a developer does it for the first time, they can be taught this wonderful bit knowledge and forever be that much better a dev :)

**Comment by Pushplaybang** [comment]
@SkippyZA @ShaunEgan  - I completely agree with you both, but I wonder if it's not too idealistic.  We do have a number of juniors on the team, we will soon have interns, and we do often work with a number of external devs, and while in principle I'm in agreement, is it not more pragmatic to simply place constraints that avoid the situation altogether and skip any back and forth later on?  

I'm skeptical as to the cost of having a little 'bloat' in the .gitignore vs anyone's man hours.

**Comment by Pushplaybang** [comment]
Any further thoughts on this @ShaunEgan @SkippyZA ?  @Nona-Creative/nonabots   lets get this moving?

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,49 @@
+### OSX ###
+.DS_Store
+*/.DS_Store
+.DS_Store?
+.AppleDouble
+.LSOverride
+
+# Icon must end with two \r
+Icon
```
Comment (SkippyZA) [inline .gitignore]: This seems a weird one to me. Will we not land up with issues when someone wants to commit a directory 'Icon'? What generates this that we don't want to commit it?

**Comment by SkippyZA** [comment]
Other than my outstanding question, this looks ok. Can't fault it other than my preference.

**Comment by Pushplaybang** [comment]
@Nona-Creative/nanobots @royscheeren @leogopal @RollYourOwnEd @JamesGarethSmith  do we have any objections / opinions / input on this?  Would really appreciate your thoughts on how you'd all like this to be, before we finalize it.

**Comment by Pushplaybang** [comment]
@SkippyZA  agreed  - I'm happy either way, if you would like to slim it down, please pull and do so. 

Keen to get this done. :)

**Comment by JamesGarethSmith** [comment]
I don't think any "bloat" from having a few extra lines in a `.gitignore` outweighs the ease of getting people going on a project. 

The most important thing is to allow new people (whether to the project, the company or even development itself) to get up and running as quickly as possible. The fewer things they have to do, outside of cloning the repo, the better.

I would be happy to have any OS and popular editor artefacts in the `.gitignore`.

**Comment by Pushplaybang** [comment]
@JamesGarethSmith  - I agree. @darrynten  - thoughts?

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+*.cab
+*.msi
+*.msm
+*.msp
+
+### Project Specific ###
+# Dependency directories
+node_modules
+.sass-cache
+
```
Comment (darrynten) [inline .gitignore]:
```
.npm
npm-debug.log
.bowerrc
bower_components
*.orig
.build
.idea/
```

**Comment by darrynten** [comment]
Needs:

```
.npm
npm-debug.log
.bowerrc
bower_components
*.orig
.build
.idea/
```

**Comment by SkippyZA** [comment]
I think lets just merge these in for a base and then make requests/issues against that

**Pull request merged**